lib/src/chunked_coding/decoder.dart

```diff
@@ -33,16 +33,20 @@ class ChunkedCodingDecoder extends Converter<List<int>, List<int>> {
 /// A conversion sink for the chunked transfer encoding.
 class _Sink extends ByteConversionSinkBase {
   /// The underlying sink to which decoded byte arrays will be passed.
-  final Sink<List<int>> _sink;
+  late final Sink<List<int>> _sink;
 
   /// The current state of the sink's parsing.
   var _state = _State.boundary;
 
   /// The size of the chunk being parsed, or `null` if the size hasn't been
   /// parsed yet.
-  int _size;
+  late int _size;
 
-  _Sink(this._sink);
+  _Sink(Sink<List<int>>? sink) {
```

Comment:
Why does this need to be nullable?

Reply:
Because of https://github.com/dart-lang/http_parser/blob/2a62c79d46e9418ec9860e4d265993c009816777/lib/src/chunked_coding/decoder.dart#L21 🤷

Reply:
How about `_Sink(StreamController())`?

Reply:
sure? unneeded allocation. But less `late` / `null` magic. 🤷

Reply:
Yeah, I think most likely what this needs is a refactoring to avoid needing to use a "Sink" that's not _actually_ a sink for this purpose. I'd be OK with either approach until then.